src/test/ui/issues/issue-23046.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0282]: type annotations needed
+error[E0282]: type annotations needed in `Expr<'_, _>`
```

Comment:
Is there a way we can replace the `_` with the type parameter name? It's confusing referring to a parameter name without it appearing in the type we print. (The user could work it out, but without being explicit, I'm not sure this is much more helpful.)

Reply:
If we use the type parameter name it might be confused for an actual type... maybe we could do `?T` -- but idk if people will understand the inference variable notation... `_` might actually be best here because people already have an intuition from `Vec<_>` and such of `_` being a placeholder.

Reply:
I think if we say "type parameter `VAR`", then it's clear that `VAR` is a type parameter.

Reply:
Sure, that's true.

Reply:
I just managed to add a hack to `ty::fmt::FmtPrinter` that gets the type arg name, I'll push it soon. I'm also glad that none of you know where I live so you can't come and find me in the middle of the night 😬

Reply:
> [...] none of you know where I live [...]

😈 🤔 💡 🗺 🔍

Reply:
[Voilá](https://github.com/rust-lang/rust/pull/61361/commits/82718cf1e16c4fd3ffc8fc6f63b0e019d5c58098)